app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,72 @@
+<% if notice.present? %>
+  <p><%= notice %></p>
+<%end %>
+<div class="p-5">
+  <div>
+  </div>
+  <h1 class="font-bold text-3xl text-center mb-8"><%= @recipe.name %></h1>
+  <div class="flex justify-around">
+    <div class="flex flex-col">
+      <h3 class="text-xl my-3">Preparation time: <%= @recipe.preparation_time %></h3>
+      <h3 class="text-xl my-3">Cooking time: <%= @recipe.cooking_time %></h3>
+      <p> <%= @recipe.description %></p>
+    </div>
+    
+      <% if @recipe.public %>
+      <div class="flex">
+        <div class="w-11 h-6 flex justify-end items-center border-1 rounded-full bg-blue-600 pr-[1.5px]">
+          <div class="w-5 h-5 rounded-full bg-white"></div>
+        </div>
+        <span class="ml-3 text-sm font-medium text-gray-400 dark:text-gray-500">Public</span>
+        </div>
+        <%else%>
+        <div class="flex">
+          <div class="w-11 h-6 flex items-center border-1 rounded-full bg-gray-500">
+            <div class="w-5 h-5 rounded-full bg-white"></div>
+          </div>
+          <span class="ml-3 text-sm font-medium text-gray-400 dark:text-gray-500">private</span>
+        </div>
+      <% end %>
+    </div>
+  </div>
+
+  <div class="flex justify-around">
+    <% if @recipe.user == current_user %>
+      <%= link_to "Generate shoping list", general_shopping_list_path(@recipe.id), class: "border px-2 py-2 font-bold"%>
```

Comment:
- [ ] When the "Generate shopping list" link is clicked, it is expected that: 
1. A page showing the list of food that is missing for all recipes of the logged-in user is rendered.
2. It should count the total food items and total price of the missing food.
However, none of that currently happens instead the following page is shown as you can see below 👇 

![image](https://user-images.githubusercontent.com/46839548/208243871-eb11c99d-7e33-47dc-9675-e1c23786665a.png)


**This is what your general shopping list should look like** 👇
![image](https://user-images.githubusercontent.com/46839548/208243991-6a109b48-00c2-4d0f-92a8-45164c1ff97a.png)

Reply:
Hey guys, once I click on the "Generate shopping list" it displays the public recipes page even though the URL path is for the shopping list page. Kindly fix this issue. You can refer to [this video](https://www.loom.com/share/6b6b8f9a0d8c469c98e5a2a0b6f36060) where I demonstrated the issue.